src/bindings/fluid_filerenderer.c

```diff
@@ -289,7 +289,32 @@ new_fluid_file_renderer(fluid_synth_t *synth)
         goto error_recovery;
     }
 
+#ifdef _WIN32
+    if (0 == FLUID_STRCMP("-", filename))
+        dev->sndfile = sf_open(filename, SFM_WRITE, &info); // pipe open
+    else
+    {
+        int u16_count;
+        LPWSTR filename_wchar;
+        dev->sndfile = NULL;
+        
+        // utf-8 filename to utf-16 filename_wchar
+        if (1 > (u16_count = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, filename, -1, NULL, 0)))
+            FLUID_LOG(FLUID_ERR, "Failed to convert UTF8 string to wide char string");
+        else if (NULL == (filename_wchar = (LPWSTR)FLUID_ARRAY(WCHAR, u16_count)))
+            FLUID_LOG(FLUID_ERR, "Out of memory");
+        else
+        {
+            if (u16_count != MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, filename, -1, filename_wchar, u16_count))
+                FLUID_LOG(FLUID_ERR, "Failed to convert UTF8 string to wide char string");
+            else
+                dev->sndfile = sf_wchar_open(filename_wchar, SFM_WRITE, &info); // file open
+            FLUID_FREE(filename_wchar);
+        }
+    }
+#else
     dev->sndfile = sf_open(filename, SFM_WRITE, &info);
+#endif
```

Comment:
in http://libsndfile.github.io/libsndfile/api.html#open : 
> The sf_open() function opens the sound file at the specified path. The filename is byte encoded, but may be utf-8 on Linux, while on Mac OS X it will use the filesystem character set. On Windows, there is also a Windows specific sf_wchar_open() that takes a UTF16_BE encoded filename.

It is quite absurd that `libsndfile` does not allow UTF-8 file names in Windows like it does on Linux. It may be acceptable this workaround in Fluidsynth, but this should be fixed upstream (`libsndfile`) as well.

Even more funny: in [1.1.0](https://github.com/libsndfile/libsndfile/releases/tag/1.1.0) you may find this change:

>Use UTF-8 as internal path encoding on Windows platform.

>This is an internal change to unify and simplify the handling of file paths.

>On the Windows platform, the file path is always converted to UTF-8 and converted to UTF-16 only for calls to WinAPI functions.

> The behavior of the functions for opening files on other platforms does notchange.